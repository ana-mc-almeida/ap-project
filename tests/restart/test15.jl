include("common.jl")

handling(DivisionByZero => (x) -> if available_restart(:return_zero)
    invoke_restart(:return_zero)
end) do
    handling(DivisionByZero =>
        (c) -> println("I saw a division by zero")) do
        println(with_restart(:return_zero => () -> 0) do
            reciprocal(0)
        end)
        println(with_restart(:return_zero => () -> 0) do
            reciprocal(0)
        end)
    end
    println(with_restart(:return_zero => () -> 0) do
        reciprocal(0)
    end)
end
