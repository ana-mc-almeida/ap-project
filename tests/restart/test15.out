I saw a division by zero
0
I saw a division by zero
0
0